cpp/velox/utils/Common.cc

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Common.h"
+
+namespace gluten {
+
+std::unique_ptr<re2::RE2> compilePattern(const std::string& pattern) {
+  return std::make_unique<re2::RE2>(re2::StringPiece(pattern), RE2::Quiet);
```

Comment:
nit: add a check whether the pattern semantic diff between re2 and java.util.regex. Such as https://github.com/facebookincubator/velox/blob/main/velox/functions/sparksql/RegexFunctions.cpp#L26

Reply:
Hi, @jackylee-ch ,The check for regex semantic diff on [RegexFunctions.cpp#L26] maybe not enough, do you have any idea for diff check?
These are some other diff case：
'|' 
'.'
'.*'
'\\.*'
'\\?*'